lib/ecto/type.ex

```diff
@@ -1143,4 +1154,11 @@ defmodule Ecto.Type do
 
   defp pad_usec(%{microsecond: {microsecond, _}} = struct),
     do: %{struct | microsecond: {microsecond, 6}}
+
+  defp check_utc_timezone!(%{time_zone: "Etc/UTC"} = datetime, _kind), do: datetime
```

Comment:
Minor nitpick: should it be `check_utc_time_zone!` so we use “time zone” separated everywhere?